### PR TITLE
[Fix/bn] Fix batchnormalize layer @open sesame 10/29 10:43

### PR DESCRIPTION
**Pull request by zhoonit**
Fix batchnormalizationLayer that epsilon is included toward saving
moving variacne that could lead to long term inaccuracy

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

### CODE REVIEW

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: nntrainer/src/bn_layer.cpp [hunk 1]
```diff
@@ -127,13 +127,14 @@ sharedConstTensors BatchNormalizationLayer::forwarding(sharedConstTensors in) {
...
 
     cvar = deviation.pow(2.0f).average(axes_to_reduce);
-    cvar.add_i(epsilon);
 
     mu.multiply_i(momentum);
     mu.add_i(cmu, 1 - momentum);
     var.multiply_i(momentum);
     var.add_i(cvar, 1 - momentum);
 
+    cvar.add_i(epsilon);
```
Comment (kparichay) [inline nntrainer/src/bn_layer.cpp:136]: This can avoid long-term inaccuracy but does cause short term inaccuracy.
If trained for only 1 iteration where cvar is 0, this change would make var 0 causing UB in evaluation mode as var remained zero.
If you are planning on doing this, you must add epsilon to var (line 144).
I urge to look at pytorch/tensorflow implementation style for this in order to avoid any differences or error cases as they have been tried and tested by a lot of people.

**Review by kparichay — CHANGES_REQUESTED** [review]
This can lead to UB.

File: nntrainer/src/bn_layer.cpp (hunk 1 above)
Comment (zhoonit) [reply nntrainer/src/bn_layer.cpp:136]: Good point. Checked bn implmentation in tf and added epsilon to the inference mode 👍

**Review by kparichay — APPROVED** [review]
LGTM now
